src/server/services/generate-arpa-report.js

```diff
@@ -28,144 +54,422 @@ async function generateReportName (periodId) {
-      // TODO: Handle matching records
-      default:
-        return null
-    }
-  }).filter(isNotNull)
+  return records
+    .filter(isProjectRecord)
+    .map(record => {
+      const detailedEcCode = getDetailedEcCode(record)
+      switch (detailedEcCode) {
+        case '5.19':
+        case '5.20':
+        case '5.21': {
+          // TODO
+          return null
+        }
+        default:
+          return null
+      }
+    })
+    .filter(isNotNull)
 }
 
 async function generateProjectBaseline (records) {
-  return records.map(record => {
-    log('record.type', record.type)
-    switch (record.type) {
-      case 'ec 1 - public health': {
-        return [
-          null, // first col is blank
-          '1-Public Health',
-          record.subcategory,
-          record.content.Name,
-          record.content.Project_Identification_Number__c,
-          record.content.Completion_Status__c,
-          record.content.Adopted_Budget__c,
-          record.content.Total_Obligations__c,
-          record.content.Total_Expenditures__c,
-          record.content.Current_Period_Obligations__c,
-          record.content.Current_Period_Expenditures__c,
-          record.content.Does_Project_Include_Capital_Expenditure__c,
-          record.content.Total_Cost_Capital_Expenditure__c,
-          record.content.Type_of_Capital_Expenditure__c,
-          record.content.Type_of_Capital_Expenditure_Other__c,
-          record.content.Capital_Expenditure_Justification__c,
-          record.content.Project_Description__c,
-          record.content.Program_Income_Earned__c,
-          record.content.Program_Income_Expended__cs,
-          record.content.Primary_Project_Demographics__c,
-          record.content.Primary_Project_Demographics_Explanation__c,
-          record.content.Secondary_Project_Demographics__c,
-          record.content.Secondary_Proj_Demographics_Explanation__c,
-          record.content.Tertiary_Project_Demographics__c,
-          record.content.Tertiary_Proj_Demographics_Explanation__c,
-          record.content.Structure_Objectives_of_Asst_Programs__c,
-          record.content.Recipient_Approach_Description__c
-        ]
+  return records
+    .filter(isProjectRecord)
+    .map(record => {
+      const detailedEcCode = getDetailedEcCode(record)
+      switch (detailedEcCode) {
+        case '1.1':
+        case '1.2':
+        case '1.3':
+        case '1.4':
+        case '1.5':
+        case '1.6':
+        case '1.7':
+        case '1.10':
+        case '1.11':
+        case '1.12':
+        case '1.13':
+        case '1.14':
+        case '2.9':
+        case '2.10':
+        case '2.11':
+        case '2.12':
+        case '2.13':
+        case '2.15':
+        case '2.16':
+        case '2.17':
+        case '2.18':
+        case '2.19':
+        case '2.20':
+        case '2.21':
+        case '2.22':
+        case '2.23':
+        case '2.28':
+        case '2.35':
+        case '2.37': {
```

Comment:
One thought:

It occurred to me that this list should logically be the inverse of any records matched by any other template.  It might be nice to think about rewriting the way we delegate records so that one record can only ever be assigned to a single export template.

However, I wasn't quite confident enough to tackle that now.

(a) It's not obvious to me what a simple algorithm that accomplishes this would be.
(b) Maybe that invariant won't hold (for example, what if we want automatically calculate project expenditure totals?)

Reply:
in python i would do it with sets. so, for instance:
```python
all_codes = set(('1.1', '1.2', '2.5', '5.19', '5.20'))
code_to_output = {
   'generateProject519521': set('5.19', '5.20')
 }
all_assigned = reduce(lambda s1, s2: s1 | s2, code_to_output.values())
code_to_output['baseline'] = all_codes - all_assigned
```